<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Camera topic and weights, config and classes files -->
    <arg name="image_topic"	                default="/cv_camera_node/image_rect_color"/>
    <arg name="frame_id"               		default="camera_link"/>
    <arg name="weights_name"	            default="chess_v2.pt"/>

    <!-- Published topics -->
    <arg name="publish_image"	            default="true"/>
    <arg name="detected_objects_topic"      default="detected_objects_in_image"/>
    <arg name="detections_image_topic"      default="detections_image_topic"/>

    <!-- Detection confidence -->
    <arg name="confidence"                  default="0.30"/>

    <!-- Node -->
    <node name="detector_manager" pkg="yolov5_pytorch_ros" type="detect.py" output="screen" respawn="true">
        <param name="image_topic"               value="$(arg image_topic)"/>
        <param name="frame_id"               		value="$(arg frame_id)"/>
        <param name="weights_name"              value="$(arg weights_name)"/>
        <param name="publish_image"             value="$(arg publish_image)"/>
        <param name="detected_objects_topic"    value="$(arg detected_objects_topic)"/>
        <param name="detections_image_topic"    value="$(arg detections_image_topic)"/>
        <param name="confidence"                value="$(arg confidence)"/>
    </node>
</launch>
